tests/suite/upstream_settings_test.go

```diff
@@ -0,0 +1,414 @@
+							Upstreams: []string{"default_coffee_80"},
+							File:      "nginx.conf",
+						},
+						{
+							Directive: "keepalive_timeout",
+							Value:     "5h",
+							Upstreams: []string{"default_coffee_80"},
+							File:      "nginx.conf",
+						},
+					}),
+				)
+			})
+
+			Expect(resourceManager.DeleteFromFiles(files, namespace)).To(Succeed())
+		})
+		Specify("configuring overlapping settings, the policy created first wins", func() {
+			files := []string{"upstream-settings-policy/valid-usps-first-wins.yaml"}
+			Expect(resourceManager.ApplyFromFiles(files, namespace)).To(Succeed())
+
+			nsname := types.NamespacedName{Name: "coffee-svc-usp-1", Namespace: namespace}
+			Expect(waitForUSPolicyStatus(nsname, metav1.ConditionTrue, v1alpha2.PolicyReasonTargetNotFound)).To(Succeed())
+
+			nsname = types.NamespacedName{Name: "coffee-svc-usp-2", Namespace: namespace}
+			Expect(waitForUSPolicyStatus(nsname, metav1.ConditionTrue, v1alpha2.PolicyReasonTargetNotFound)).To(Succeed())
+
+			Context("verify working traffic", func() {
+				It("should return a 200 response for HTTPRoute", func() {
+					port := 80
+					if portFwdPort != 0 {
+						port = portFwdPort
+					}
+					baseURL := fmt.Sprintf("http://cafe.example.com:%d%s", port, "/coffee")
+
+					Eventually(
+						func() error {
+							return expectRequestToSucceed(baseURL, address, "URI: /coffee")
+						}).
+						WithTimeout(timeoutConfig.RequestTimeout).
+						WithPolling(500 * time.Millisecond).
+						Should(Succeed())
+				})
+			})
+
+			Context("nginx directives", func() {
+				var conf *framework.Payload
+
+				BeforeAll(func() {
+					podNames, err := framework.GetReadyNGFPodNames(k8sClient, ngfNamespace, releaseName, timeoutConfig.GetTimeout)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(podNames).To(HaveLen(1))
+
+					ngfPodName := podNames[0]
+
+					conf, err = resourceManager.GetNginxConfig(ngfPodName, ngfNamespace)
+					Expect(err).ToNot(HaveOccurred())
+				})
+
+				// TODO: important
+				// The directive file and field value need to be updated based on the
+				// implementation of the UpstreamSettingsPolicy and how they are specified in the config files.
+				DescribeTable("are set properly for",
+					func(expCfgs []framework.ExpectedNginxField) {
+						for _, expCfg := range expCfgs {
+							Expect(framework.ValidateNginxFieldExists(conf, expCfg)).To(Succeed())
+						}
+					},
+					Entry("HTTPRoute", []framework.ExpectedNginxField{
+						{
+							Directive:             "zone default_coffee_80",
+							Value:                 "128k",
+							Upstreams:             []string{"default_coffee_80"},
+							File:                  "nginx.conf",
+							ValueSubstringAllowed: true,
+						},
+					}),
+				)
+			})
+			Expect(resourceManager.DeleteFromFiles(files, namespace)).To(Succeed())
+		})
+	})
+
+	When("UpstreamSettings Policy is invalid", func() {
+		Specify("if service mentioned in the target ref is not present", func() {
```

Comment:
```suggestion
		Specify("no conditions will be added to its status", func() {
```

We will not write status to a USP if it targets a service that doesn't exist.